onnxruntime/core/providers/trt/trt_execution_provider.h

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <ctime>
+#include "core/framework/op_kernel.h"
+#include "NvInfer.h"
+#include "NvOnnxParser.h"
+
+namespace onnxruntime {
+
+static const int kBatchSize = 1;
+static const int max_batch_size = 1;
+static const int max_workspace_size = 1 << 30;
+
+#define CHECK(status)                       \
```

Comment:
>CHECK [](start = 8, length = 5)

Should a #define with such a general name should be in the .h file?

Also the error message says 'Cuda failure'. Is that correct?

Reply:
The CHECK is for cuda APIs. I will change the name to CUDA_CHECK.